zephyr/lvgl_shell.c

```diff
@@ -0,0 +1,212 @@
+				    lv_monkey_get_enable(lvgl_monkeys[i]) ? "yes" : "no");
+		}
+	}
+}
+#endif
+
+static int cmd_lvgl_stats(const struct shell *sh, size_t argc, char *argv[])
+{
+	ARG_UNUSED(argc);
+	ARG_UNUSED(argv);
+
+	shell_help(sh);
+	return SHELL_CMD_HELP_PRINTED;
+}
+
+static int cmd_lvgl_stats_memory(const struct shell *sh, size_t argc, char *argv[])
+{
+#ifdef CONFIG_LV_Z_MEM_POOL_SYS_HEAP
+	bool dump_chunks = false;
+
+	if (argc == 2) {
+		if (strcmp(argv[1], "-c") == 0) {
+			dump_chunks = true;
+		} else {
+			shell_error(sh, "unsupported option %s", argv[1]);
+			shell_help(sh);
+			return SHELL_CMD_HELP_PRINTED;
+		}
+	}
+
+	lvgl_print_heap_info(dump_chunks);
+	return 0;
+#else
+	ARG_UNUSED(argc);
+	ARG_UNUSED(argv);
+	shell_error(sh, "Set CONFIG_LV_Z_MEM_POOL_SYS_HEAP to enable memory statistics support.");
+	return -ENOTSUP;
+#endif
+}
+
+static int cmd_lvgl_monkey(const struct shell *sh, size_t argc, char *argv[])
+{
+	ARG_UNUSED(argc);
+	ARG_UNUSED(argv);
+
+#ifdef CONFIG_LV_USE_MONKEY
+	dump_monkey_info(sh);
+	shell_print(sh, "");
+	shell_help(sh);
+	return SHELL_CMD_HELP_PRINTED;
+#else
+	shell_error(sh, "Set CONFIG_LV_USE_MONKEY to enable monkey testing support.");
+	return -ENOTSUP;
+#endif
+}
+
+static int cmd_lvgl_monkey_create(const struct shell *sh, size_t argc, char *argv[])
+{
+#ifdef CONFIG_LV_USE_MONKEY
+	bool created_monkey = false;
+	lv_monkey_config_t default_config;
+	lv_monkey_config_init(&default_config);
+
+	if (argc == 2) {
+		if (lvgl_monkey_indev_from_string(argv[1], &default_config.type) < 0) {
+			shell_error(sh, "Invalid monkey input device %s", argv[1]);
+			shell_help(sh);
+			return SHELL_CMD_HELP_PRINTED;
+		}
+	}
+
+	for (size_t i = 0; i < CONFIG_LV_Z_MAX_MONKEY_COUNT; i++) {
+		if (lvgl_monkeys[i] == NULL) {
+			lvgl_monkeys[i] = lv_monkey_create(&default_config);
+			lv_monkey_set_enable(lvgl_monkeys[i], true);
+			created_monkey = true;
+			break;
+		}
+	}
+
+	if (!created_monkey) {
+		shell_error(sh, "Error creating monkey instance");
+	}
+
+	dump_monkey_info(sh);
+#else
+	ARG_UNUSED(argc);
+	ARG_UNUSED(argv);
+	shell_error(sh, "Set CONFIG_LV_USE_MONKEY to enable monkey testing support.");
+	return -ENOTSUP;
+#endif
+	return 0;
+}
+
+static int cmd_lvgl_monkey_set(const struct shell *sh, size_t argc, char *argv[])
+{
+#ifdef CONFIG_LV_USE_MONKEY
+	int index;
+
+	index = atoi(argv[1]);
+	if (index < 0 || index >= CONFIG_LV_Z_MAX_MONKEY_COUNT || lvgl_monkeys[index] == NULL) {
+		shell_error(sh, "Invalid monkey index");
+		return -ENOEXEC;
+	}
+
+	if (strcmp(argv[2], "active") == 0) {
+		lv_monkey_set_enable(lvgl_monkeys[index], true);
+	} else if (strcmp(argv[2], "inactive") == 0) {
+		lv_monkey_set_enable(lvgl_monkeys[index], false);
+	} else {
+		shell_error(sh, "Invalid monkey state %s", argv[2]);
+		shell_help(sh);
+		return SHELL_CMD_HELP_PRINTED;
+	}
+
+	dump_monkey_info(sh);
+#else
+	shell_error(sh, "Set CONFIG_LV_USE_MONKEY to enable monkey testing support.");
+	return -ENOTSUP;
+#endif
```

Comment:
why not drop the command entirely if not enabled?

Reply:
I thought about it, but I was not sure what the preferred method was. I'll adjust it  :^)